app/scripts/directives/bindService.js

```diff
@@ -168,6 +169,8 @@
         });
       }
       else {
+        ctrl.bindType = 'application';
+        ctrl.appToBind = ctrl.target;
```

Comment:
Lines `173` and `184` seem to be the same & I believe one can be removed. 

I'd actually prefer to see some of this logic broken out into separate functions.  Its a gnarly big if/else & there might be some long term benefits to breaking it down into smaller pieces.